pkg/oci/load_balancer_util_test.go

```diff
@@ -661,3 +661,117 @@ func TestGetLoadBalancerName(t *testing.T) {
 		})
 	}
 }
+
+func TestGetLoadBalancerSourceRanges(t *testing.T) {
+	testCases := map[string]struct {
+		config       *Config
+		service      *api.Service
+		sourceRanges []string
+		err          error
+	}{
+		"not-specified": {
+			config:       &Config{},
+			service:      &api.Service{},
+			sourceRanges: []string{"0.0.0.0/0"},
+			err:          nil,
+		},
+		"spec-field-basic": {
+			config: &Config{},
+			service: &api.Service{
+				Spec: api.ServiceSpec{
+					LoadBalancerSourceRanges: []string{"10.0.0.0/24"},
+				},
+			},
+			sourceRanges: []string{"10.0.0.0/24"},
+			err:          nil,
+		},
+		"annotation-basic": {
+			config: &Config{},
+			service: &api.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						api.AnnotationLoadBalancerSourceRangesKey: "10.0.0.0/24",
+					},
+				},
+			},
+			sourceRanges: []string{"10.0.0.0/24"},
+			err:          nil,
+		},
+		"spec-field-and-annotation": {
+			config: &Config{},
+			service: &api.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						api.AnnotationLoadBalancerSourceRangesKey: "10.0.0.0/16",
+					},
+				},
+				Spec: api.ServiceSpec{
+					LoadBalancerSourceRanges: []string{"10.1.0.0/16"},
+				},
+			},
+			sourceRanges: []string{"10.1.0.0/16"}, // field supersedes annotation
+			err:          nil,
+		},
+		"empty-default": {
+			config: &Config{
+				LoadBalancer: LoadBalancerConfig{
+					DefaultSourceCIDRs: []string{},
+				},
+			},
+			service:      &api.Service{},
+			sourceRanges: []string{"0.0.0.0/0"},
+			err:          nil,
+		},
+		"default-only": {
+			config: &Config{
+				LoadBalancer: LoadBalancerConfig{
+					DefaultSourceCIDRs: []string{"10.0.0.0/24"},
+				},
+			},
+			service:      &api.Service{},
+			sourceRanges: []string{"10.0.0.0/24"},
+			err:          nil,
+		},
+		"default-overridden-field": {
+			config: &Config{
+				LoadBalancer: LoadBalancerConfig{
+					DefaultSourceCIDRs: []string{"10.1.0.0/24"},
+				},
+			},
+			service: &api.Service{
+				Spec: api.ServiceSpec{
+					LoadBalancerSourceRanges: []string{"10.0.0.0/24"},
+				},
+			},
+			sourceRanges: []string{"10.0.0.0/24"},
+			err:          nil,
+		},
+		"default-overridden-annotation": {
+			config: &Config{
+				LoadBalancer: LoadBalancerConfig{
+					DefaultSourceCIDRs: []string{"10.1.0.0/24"},
+				},
+			},
+			service: &api.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						api.AnnotationLoadBalancerSourceRangesKey: "10.0.0.0/24",
+					},
+				},
+			},
+			sourceRanges: []string{"10.0.0.0/24"},
+			err:          nil,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			sourceRanges, err := getLoadBalancerSourceRanges(tc.config, tc.service)
+			if err != tc.err {
+				t.Errorf("getLoadBalancerSourceRanges() returned error %v, expected %v", err, tc.err)
```

Comment:
This should be a `Fatalf` since if an error is returned you can expect the latter to not match as well. `t.Error` still continues the test run for this case.